java/src/main/native/src/TableJni.cpp

```diff
@@ -263,16 +263,24 @@ class native_arrow_ipc_writer_handle final {
       // empty table, so need to write an empty batch explicitly.
       // For more please see https://issues.apache.org/jira/browse/ARROW-17912.
       auto empty_batch = arrow::RecordBatch::MakeEmpty(arrow_tab->schema());
-      writer->WriteRecordBatch(*(*empty_batch));
+      if (!writer->WriteRecordBatch(*(*empty_batch)).ok()) {
+        throw std::runtime_error("writer failed to write batch");
```

Comment:
For here and elsewhere in this file, it would be good to add the `Status::ToString` result to the exception message thrown so that, if this does occur in practice, it's not a complete mystery why.

Reply:
Done.